src/main/java/com/synopsys/integration/detect/ApplicationUpdater.java

```diff
@@ -247,25 +254,31 @@ private boolean checkInstallationDir(Path path) throws NoSuchFileException, Acce
     }
     
     private void checkEnvironmentProperties() {
-        this.blackduckHost = System.getenv(SYS_ENV_PROP_BLACKDUCK_URL);
-        final String httpsHost = System.getenv(SYS_ENV_PROP_PROXY_HTTPS_HOST);
-        final String httpHost = System.getenv(SYS_ENV_PROP_PROXY_HTTP_HOST);
+        this.blackduckHost = utility.getSysEnvProperty(SYS_ENV_PROP_BLACKDUCK_URL);
+        final String httpsHost = utility.getSysEnvProperty(SYS_ENV_PROP_PROXY_HTTPS_HOST);
+        final String httpHost = utility.getSysEnvProperty(SYS_ENV_PROP_PROXY_HTTP_HOST);
         
         if (httpsHost != null && StringUtils.isNotBlank(httpsHost)) {
             /* HTTPS Proxy System Environment settings are preferred. */
-            proxyProperties.put(ARG_PROXY_PORT, httpsHost);
-            proxyProperties.put(ARG_PROXY_PORT, System.getenv(SYS_ENV_PROP_PROXY_HTTPS_PORT));
-            proxyProperties.put(ARG_PROXY_PORT, System.getenv(SYS_ENV_PROP_PROXY_HTTPS_USERNAME));
-            proxyProperties.put(ARG_PROXY_PORT, System.getenv(SYS_ENV_PROP_PROXY_HTTPS_PASSWORD));
+            proxyPropertiesPut(ARG_PROXY_HOST, httpsHost);
+            proxyPropertiesPut(ARG_PROXY_PORT, utility.getSysEnvProperty(SYS_ENV_PROP_PROXY_HTTPS_PORT));
+            proxyPropertiesPut(ARG_PROXY_USERNAME, utility.getSysEnvProperty(SYS_ENV_PROP_PROXY_HTTPS_USERNAME));
+            proxyPropertiesPut(ARG_PROXY_PASSWORD, utility.getSysEnvProperty(SYS_ENV_PROP_PROXY_HTTPS_PASSWORD));
         } else if (httpHost != null && StringUtils.isNotBlank(httpHost)) {
             /* If not HTTPS, then HTTP is fine. But Log a warning. */
             logger.warn("{} Use of HTTP instead of HTTPS for Proxy system environment properties was found. "
                     + "It is strongly recommended to use HTTPS over HTTP for increased security."
                     + "Detect will continue self update check with the configured HTTP settings.");
```

Comment:
What is the usecase of '{}' in `logger.warn("{}...."); `? You are not logging any varibale, right?

Reply:
This line printing empty curly-braces like this:
`2023-05-10 22:27:21 BDT WARN  [Test worker] --- {} Use of HTTP instead of HTTPS for Proxy system environment properties was found. It is strongly recommended to use HTTPS over HTTP for increased security.Detect will continue self update check with the configured HTTP settings.
2023-05-10 22:27:21 BDT WARN  [Test worker] --- {} Use of HTTP instead of HTTPS for Proxy system environment properties was found. It is strongly recommended to use HTTPS over HTTP for increased security.Detect will continue self update check with the configured HTTP settings.`

Reply:
Fixed, thanks!